ostd/src/arch/x86/mod.rs

```diff
@@ -177,7 +178,48 @@ fn has_avx512() -> bool {
 }
 
 pub(crate) fn enable_cpu_features() {
-    use x86_64::registers::{control::Cr4Flags, model_specific::EferFlags, xcontrol::XCr0Flags};
+    use x86::cpuid;
+    use x86_64::registers::{
+        control::{Cr0Flags, Cr4Flags},
+        model_specific::EferFlags,
+        xcontrol::XCr0Flags,
+    };
+
+    CPU_FEATURES.call_once(|| {
```

Comment:
Move `CPU_FEATURES` to the root module of `x86` as every module may query CPU features.